### ISSUE TITLE
Bold breaks open graph metadata due to new line

### ISSUE THREAD

**Issue by dylanh724** [sphinx-doc/sphinx#12777]
### Describe the bug

In RST after my h1 in my body, I have:

```rst
**Foo** Bar lorum ipsum
```

In HTML, this generates as:
```html
<strong>Foo</strong>
Bar lorum ipsum
```

**Notice the line break.** For metadata parsing like open graph, this is pulled as `FooBar lorum ipsum` vs the expected `Foo Bar lorum ipsum`.

**Expected:**

Bold should not cause a line break. The resulting output is expected to be:

```html
<strong>Foo</strong> Bar lorum ipsum
```

### How to Reproduce

Have this at the top of your `index rst`:

```rst
**Foo** Bar lorum ipsum
```

Then implement open graph to auto-pull from the descr. I personally use `sphinxext-opengraph>=0.9.1` and host it on RTD.

### Environment Information

```text
Platform:              win32; (Windows-10-10.0.22631-SP0)
Python version:        3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)])
Python implementation: CPython
Sphinx version:        7.4.7
Docutils version:      0.20.1
Jinja2 version:        3.1.4
Pygments version:      2.18.0
```

### Sphinx extensions

```python
sphinxext-opengraph>=0.9.1
sphinx_rtd_theme==2.0
sphinx==7.4.7
```

### Additional context

How to keep `<strong>` inline? That line break seems unnecessary.

**Comment by AA-Turner** [comment]
Please confirm that this isn't an issue with sphinxext-opengraph first.

A

**Comment by dylanh724** [comment]
> Please confirm that this isn't an issue with sphinxext-opengraph first.
> 
> A

Any metadata scraper should do this since if you inspect the html output (demonstrated above), `<strong>` creates an unnecessary line break (demonstrated above).

**Comment by dylanh724** [comment]
tl;dr:

## Output

```html
<strong>Foo</strong>
Bar lorum ipsum
```

## Expected

```html
<strong>Foo</strong> Bar lorum ipsum
```

**Comment by dylanh724** [comment]
@AA-Turner Why was this closed?

**Comment by electric-coder** [comment]
Poor bug report, I don't see a new line here:

```html
<strong>Foo</strong>
Bar lorum ipsum
```

A new line would be `</br>` so where is it?

Also, here:

> In RST after my h1 in my body, 

Where's the rst? We aren't mind readers and the report doesn't provide a minimal reproducible example.

**Comment by electric-coder** [comment]
So I just tested it (using the older Sphinx v7.2.6), this rst:

```rst
########################
Glossary with some words
########################

**Foo** Bar lorum ipsum
```

Gives this HTML:

```html
<h1>Glossary with some words<a class="headerlink" href="[#glossary-with-some-words](view-source:file:///C:/JPC/Portugues/docs/build/html/glossary.html#glossary-with-some-words)" title="Link to this heading">#</a></h1>
<p><strong>Foo</strong> Bar lorum ipsum</p>
```

So there's no new line. Without providing a minimal reproducible example the only explanation is that you're encountering a behavior that's specific to [`sphinxext-opengraph`][1] and hence the question should be asked core Sphinx seems to be working correctly.

[1]: https://github.com/wpilibsuite/sphinxext-opengraph

**Comment by electric-coder** [comment]
So I just tested it (using the older Sphinx v7.2.6), this rst:

```rst
########################
Glossary with some words
########################

**Foo** Bar lorum ipsum
```

Gives this HTML:

```html
<h1>Glossary with some words<a class="headerlink" href="[#glossary-with-some-words](view-source:file:///C:/prtoject/docs/build/html/glossary.html#glossary-with-some-words)" title="Link to this heading">#</a></h1>
<p><strong>Foo</strong> Bar lorum ipsum</p>
```

So there's no new line. Without providing a minimal reproducible example the only explanation is that you're encountering a behavior that's specific to [`sphinxext-opengraph`][1] and hence the question should be asked there because core Sphinx seems to be working correctly.

[1]: https://github.com/wpilibsuite/sphinxext-opengraph
